platform/security/platform-security-core-api/src/test/java/ddf/security/PropertiesFileReaderTest.java

```diff
@@ -0,0 +1,126 @@
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.security;
+
+import static java.lang.String.format;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class PropertiesFileReaderTest {
+    @ClassRule
+    public static TemporaryFolder testFolder = new TemporaryFolder();
+
+    private static final Path NONEXISTENT_PATH = Paths.get("this", "path");
+
+    private static final int VALUES_PER_FILE = 4;
+
+    private static String propertiesTestDirectoryPath;
+
+    private static String propertiesTestSingleFilePath;
+
+    private PropertiesFileReader reader = new PropertiesFileReader();
+
+    @BeforeClass
+    public static void setup() throws Exception {
+        File propertiesTestDirectory = testFolder.newFolder("properties");
+        propertiesTestDirectoryPath = propertiesTestDirectory.getPath();
+        propertiesTestSingleFilePath = Paths.get(propertiesTestDirectoryPath, "test1.properties")
+                .toString();
+
+        File propsFile1 = new File(propertiesTestDirectory, "test1.properties");
+        File propsFile2 = new File(propertiesTestDirectory, "test2.properties");
+        File propsFile3 = new File(propertiesTestDirectory, "test3.properties");
+
+        File randomFile = new File(propertiesTestDirectory, "random.cfg");
+        assert randomFile.createNewFile();
+
+        generateAndSaveProperties(propsFile1, "test1value", VALUES_PER_FILE);
+        generateAndSaveProperties(propsFile2, "test2value", VALUES_PER_FILE);
+        generateAndSaveProperties(propsFile3, "test3value", VALUES_PER_FILE);
+
+    }
+
+    @Test
+    public void testEmptyStringForPath() throws Exception {
+        List resultsDirectory = reader.loadPropertiesFilesInDirectory("");
+        Map resultsOneFile = reader.loadSinglePropertiesFile("");
+        assertTrue(resultsDirectory.isEmpty());
+        assertTrue(resultsOneFile.isEmpty());
+    }
+
+    @Test
+    public void testNonExistentPath() throws Exception {
+        List resultsDirectory = reader.loadPropertiesFilesInDirectory(NONEXISTENT_PATH.toString());
+        Map resultsOneFile = reader.loadSinglePropertiesFile(NONEXISTENT_PATH.toString());
+        assertTrue(resultsDirectory.isEmpty());
+        assertTrue(resultsOneFile.isEmpty());
+    }
+
+    @Test
+    public void testLoadPropertiesFilesInDirectory() throws Exception {
+        List<Map<String, String>> resultsDirectory = reader.loadPropertiesFilesInDirectory(
+                propertiesTestDirectoryPath);
+        validateAllProps(resultsDirectory);
+    }
+
+    @Test
```

Comment:
❗️ We should have tests for:
* An empty file or non-existent file name
* A `null` file name
* A path that's a directory instead of a file
* A file that doesn't exist
* If possible, a file that can't be read